classes/filter_manager.php

```diff
@@ -141,6 +128,28 @@ public function get_user_filters() {
         return $filters;
     }
 
+	// Used when adding block to course
+    public function copy_default_filters_to_course() {
+        global $DB;
+
+        $course_filters = $DB->get_records('block_xp_filters', array('courseid' => $this->manager->get_courseid()));
+
+        // Only copy default filters if there are no course filters
+        if (empty($course_filters)) {
```

Comment:
I would prefer if this method didn't enforce any restriction. In my understanding, it's up to the calling code to determine whether the default rules should be added. For example, we could give the users an option to 'restore default rules'...

Moodle's coding guidelines prohibit underscores in variable names. I know...